### DISCUSSION TITLE
Usage of non-istioctl Cobra rootCmd like "mesh", "operator", "pilot-agent", etc.

### DISCUSSION THREAD

**Discussion by choigonyok** [istio/istio#48627] [Q&A] [answered]
Hi, I'm studying Istio code base. :)

I just figured out Isito uses cobra CLI software for istioctl. And I found that root commands, "istioctl" and "mesh" add a lot of same sub-commands like "ProfileCmd", "InstallCmd", "OperatorCmd" and so on..

1. Why they share same sub-commands?
2. What is usage of non-isitoctl root commands like "mesh", "operator", "pilot-agent", "pilot-discovery"?
3. Is it related with Makefile?

Thanks for your reading!

**Comment by hanxiaop** [comment, marked as answer]
1. commands are actually used in istioctl, but they may also be used by other commands like `mesh` for testing purposes.
2. They are used to build images, for example, the Istio operator, pilot, etc.
3. Makefile.core.mk
